.github/workflows/validate_release.yml

```diff
@@ -0,0 +1,62 @@
+name: Validate release
+
+on:
+  workflow_call:
+    inputs:
+      path:
+        description: 'Go mod directory (e.g. "plugins/source/aws", "plugins/destination/csv", "cli")'
+        type:        string
+        required:    true
+      runner:
+        description: 'Runner to be used (default: "ubuntu-latest")'
+        type:        string
+        required:    false
+        default:     'ubuntu-latest'
+jobs:
+  validate-release:
+    # this if check is here as we check for 'validate-release / validate-release' run
+    if:              startsWith(github.head_ref, 'release-please--branches--main--components') || github.event_name == 'push'
+    timeout-minutes: 60
+    runs-on:         ${{ inputs.runner }}
+    env:
+      CGO_ENABLED: 0
+    steps:
+    - id:  dash-path
+      run: |
+           echo "dashed=$(echo ${{ inputs.path }} | tr '/' '-')" >> $GITHUB_OUTPUT
+    - name: Checkout
+      uses: actions/checkout@v3
+    - uses: actions/cache@v3
+      with:
+        path:         |
+                      ~/.cache/go-build
+                      ~/go/pkg/mod
+        key:          ${{ runner.os }}-go-1.19.5-release-cache-${{ hashFiles(format('{0}/go.sum', inputs.path)) }}
+        # actually, this value is never ever saved, but left for consistency (renaming cache keys os out of scope)
```

Comment:
I don't understand this comment. When a cache is saved, it's saved both to `key` and each `restore-key`.
When the cache is restored, and there's a cache miss on `key` the `restore-keys` will be used in order.

Reply:
Cache isa saved _only_ for `key`.
`restore-keys` is useful to match (1 by 1 until the match is found) partially useful caches.
In this case, the `cache-hit` is `false` (as we didn't find the cache designated by `key`), so the cache will be saved in `post cache` step.

However, the `restore-keys` are just a list of possible caches that we might find a bit useful, and are usually listed from the most narrow scope to the most wide one.

Let's consider this example:
* we save cache by `key: a-b-c-d-hash(some-files)`
* we have somewhat related hashed that were saved with the same key pattern, but for different files (different file states)

Then, the following pattern can be used:

```yaml
key: a-b-c-d-hash(some-files)
restore-keys: |-
  a-b-c-d-
  a-b-c-
  a-b-
  a-
```

This allows us to restore the most matching cache entry in our store (if there's no `a-b-c-d-XYZ` entry, but we have `a-b-c-XYZ`, then this entry will be restore and the check process will stop).

Reply:
Cool, thanks for the context. So let's remove it altogether? That should simplify the workflow and remove the need for the dash logic?

Reply:
I just wanted to leave this consistent with the code I'm replacing (as all `validate-release` cache steps use `restore-keys`).

I can remove the restore part, however, the restore-keys has some benefits to be aware when we discuss cache strategy further (basically, having dirty cache is better that not to have any cache at all).

Reply:
So, should I remove this part for now and we're OK with this being inconsistent with the `validate-release` for now (although we know for sure that the `restore-keys` part doesn't have any hit due to the fact that we don't save any cache prefixed that way).
@erezrokah could you confirm?